src/commandsAndMenu.tsx

```diff
@@ -407,11 +419,24 @@ export function addCommands(
           'Encountered an error when pulling changes. Error: ',
           error
         );
-        logger.log({
-          message: trans.__('Failed to pull'),
-          level: Level.ERROR,
-          error
-        });
+
+        // Discard changes then retry pull
+        if (
+          (error as string)
```

Comment:
Error may be a `Error` object that encapsulated the `message`:

```suggestion
        const errorMsg = typeof error === 'string' ? error : error.message;
        if (
          errorMsg
```